File: Python3/Maximum Subarray.py
class Solution:
    def maxSubArray(self, nums: List[int]) -> int:
        a = 0    
        max_ = local = nums[0]
        
        for n in nums[1:]:
            sublocal = local + n
            local = n if local < 0 or sublocal < 0 else sublocal
            if local > max_:
                max_ = local
    
        return max_

File: JavaScript/Maximum Subarray.js
var maxSubArray = function(nums) {
        let a = 0    
        let max_ = nums[0]
        let local = nums[0]
        
        for (let i = 1;  i < nums.length; i++) {            
            if (local < 0 || local + nums[i] < 0) {
                local = nums[i]
            } else {
                local += nums[i]
            }
            if (local > max_) {
                max_ = local
            }
        }
        return max_
}
